src/datasets/features.py

```diff
@@ -33,18 +34,60 @@
 
 
 def string_to_arrow(type_str: str) -> pa.DataType:
-    if type_str not in pa.__dict__:
+    """
+    string_to_arrow takes a string pyarrow dtype and converts it to a pyarrow.DataType.
+
+    In effect, `dt == string_to_arrow(str(dt))`
```

Comment:
This is true only for types supported by Value. For instance decimal is not supported afaik